src/resources/ble_esp32.cc

```diff
@@ -559,6 +546,9 @@ PRIMITIVE(init) {
     }
   }
 
+  group->register_resource(gap);
```

Comment:
If the `gap` is only registered here, do we then need to deallocate it explicitly when the init-server fails (line 544/545)?

Reply:
Yeah. I believe we do need to `delete gap` before returning in line 545. Thoughts, @mikkeldamsgaard?